pkg/devfile/validate/components.go

```diff
@@ -20,12 +24,31 @@ func ValidateComponents(components []common.DevfileComponent) error {
 		return fmt.Errorf(ErrorNoComponents)
 	}
 
-	// Check if component of type container  is present
+	processedVolumes := make(map[string]bool)
+
+	// Check if component of type container is present
+	// and if volume components are unique
 	isContainerComponentPresent := false
 	for _, component := range components {
 		if component.Container != nil {
 			isContainerComponentPresent = true
-			break
+		}
+
+		if component.Volume != nil {
+			if _, ok := processedVolumes[component.Volume.Name]; !ok {
+				processedVolumes[component.Volume.Name] = true
+				if !pushtarget.IsPushTargetDocker() && len(component.Volume.Size) > 0 {
```

Comment:
shouldn't we validate negative volume size too?

Reply:
WDYM? This if condition is for checking if size is mentioned in the volume components since it is optional. I am doing len(component.Volume.Size) to check if the string is mentioned.

Values will be validated with the kube client API that i moved earlier

Reply:
ya, sorry i read `component.Volume.Size` > 0